### PR TITLE
CIL API update

### PR DESCRIPTION
**Pull request by ParnianAlimi**
Address CRYP-170. Refactor's Polymesh's use of the CIL library. Essentially removes objects to Scalar conversions.

`mercat` is the only branch that points to the cryptography's develop AFAIK, if there are others, they should cherry-pick this change as well.

The PR for changes to the CIL if you are curious: https://github.com/PolymathNetwork/cryptography/pull/85

### CODE REVIEW

File: pallets/confidential/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ sp-runtime-interface = { git = "https://github.com/paritytech/substrate", defaul
 # Crypto
 rand_core = { version = "0.5", default-features = false }
 rand = { version = "0.7.3", default-features = false, optional = true }
-cryptography = { git = "https://github.com/PolymathNetwork/cryptography.git", branch = "develop", default-features = false }
+cryptography = { git = "https://github.com/PolymathNetwork/cryptography.git", branch = "CRYP-166/take_in_slices", default-features = false }
```
Comment (ParnianAlimi) [inline pallets/confidential/Cargo.toml:33]: I will be reverting back to `develop` as soon as my other PR goes in.

File: polymesh_schema.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "types": {
         "IdentityId": "[u8; 32]",
-        "InvestorUid": "[u8; 32]",
+        "InvestorUid": "[u8; 16]",
```
Comment (maxsam4) [inline polymesh_schema.json:4]: Hmm, Is 128 bits of entropy enough for our use-case?

File: pallets/confidential/Cargo.toml (hunk 1 above)
Comment (maxsam4) [reply pallets/confidential/Cargo.toml:33]: LGTM but I'd suggest merging this after the PR in crypto repo is merged.

**Review by maxsam4 — APPROVED** [review]

File: primitives/src/investor_zkproof_data.rs [hunk 3]
```diff
@@ -37,24 +35,12 @@ impl InvestorZKProofData {
...
     /// Returns the CDD claim of the given `investor_did` and `investor_uid`.
     pub fn make_cdd_claim(investor_did: &IdentityId, investor_uid: &InvestorUid) -> CDDClaimData {
-        let investor_did = Scalar::from_bits(investor_did.to_bytes());
-        let investor_unique_id = scalar_blake2_from_bytes(&investor_uid);
-
-        CDDClaimData {
-            investor_did,
-            investor_unique_id,
-        }
+        CDDClaimData::new(investor_did.as_bytes(), &investor_uid)
```
Comment (vkomenda) [inline primitives/src/investor_zkproof_data.rs:38]: This could be camel-cased, `Cdd`.

File: polymesh_schema.json (hunk 2 above)
Comment (ParnianAlimi) [reply polymesh_schema.json:4]: This is the output of PUIS. I'm tagging @JesseALancaster to comment.

File: pallets/confidential/Cargo.toml (hunk 1 above)
Comment (ParnianAlimi) [reply pallets/confidential/Cargo.toml:33]: I'll make sure that happens.

**Review by satyamakgec — APPROVED** [review]

**Review by aafshar-poly — APPROVED** [review]
Nice job :+1:

**Pull request merged**